server.py

```diff
@@ -174,11 +175,156 @@ def new_event():
     if len(base_url) != 0:
         text = fix_gitlab_links(base_url, text)
 
-    post_text(text)
+    post_text(text, MATTERMOST_WEBHOOK_URL)
 
     return 'OK'
 
-def post_text(text):
+@app.route('/new_event_hook/<use_this_hook>', methods=['POST'])
+def new_event_hook(use_this_hook):
+    """
+    GitLab event handler, handles POST events from a GitLab project
+    """
+
+    if request.json is None:
+        print 'Invalid Content-Type'
+        return 'Content-Type must be application/json and the request body must contain valid JSON', 400
+
+    data = request.json
+    object_kind = data['object_kind']
+
+    text = ''
+    base_url = ''
+
+    if REPORT_EVENTS[PUSH_EVENT] and  object_kind == PUSH_EVENT:
```

Comment:
Sure, I can do that tonight

On Thursday, December 17, 2015, Joram Wilander <notifications@github.com>
wrote:

> In server.py
> <https://github.com/mattermost/mattermost-integration-gitlab/pull/16#discussion_r47903118>
> :
>
> > +def new_event_hook(use_this_hook):
> > +    """
> > +    GitLab event handler, handles POST events from a GitLab project
> > +    """
> > +
> > +    if request.json is None:
> > +        print 'Invalid Content-Type'
> > +        return 'Content-Type must be application/json and the request body must contain valid JSON', 400
> > +
> > +    data = request.json
> > +    object_kind = data['object_kind']
> > +
> > +    text = ''
> > +    base_url = ''
> > +
> > +    if REPORT_EVENTS[PUSH_EVENT] and  object_kind == PUSH_EVENT:
>
> This giant if statement is just a copy of the one in the other function,
> correct? Could you move it to a function that both routes call so that we
> don't have duplicate code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mattermost/mattermost-integration-gitlab/pull/16/files#r47903118>
> .
>